### PR TITLE
[Chore] Update Alpha with Master

### PR DESCRIPTION
**Pull request by stevensJourney**
This updates the alpha release branch with master. Doing this separately results in less changes for https://github.com/powersync-ja/sqlite_async.dart/pull/43.

### CODE REVIEW

File: test/native/basic_test.dart [hunk 1]
```diff
@@ -305,10 +305,11 @@ void main() {
       // 4. Now second connection is ready. Second query has two connections to choose from.
       // 5. However, first connection is closed, so it's removed from the pool.
       // 6. Triggers `Concurrent modification during iteration: Instance(length:1) of '_GrowableList'`
-      final db = await testUtils.setupDatabase(path: path, initStatements: [
+      final db = SqliteDatabase.withFactory(
```
Comment (DominicGBauer) [inline test/native/basic_test.dart:308]: why is this change needed?

**Review by DominicGBauer — APPROVED** [review]

File: test/native/basic_test.dart (hunk 1 above)
Comment (stevensJourney) [reply test/native/basic_test.dart:308]: [Discussed offline]  This is inline with how `master` currently is.
